addons/spreadsheet/static/src/chart/plugins/chart_odoo_menu_plugin.js

```diff
@@ -21,6 +24,35 @@ export default class ChartOdooMenuPlugin extends spreadsheet.CorePlugin {
             case "DELETE_FIGURE":
                 this.history.update("odooMenuReference", cmd.id, undefined);
                 break;
+            case "DUPLICATE_SHEET":
+                this.updateOnDuplicateSheet(cmd.sheetId, cmd.sheetIdTo);
+                break;
+        }
+    }
+
+    updateOnDuplicateSheet(sheetIdFrom, sheetIdTo) {
+        for (const oldChartId of this.getters.getChartIds(sheetIdFrom)) {
```

Comment:
Probably some kind of nitpick, but maybe we can have two sets (for the old and new figures) and, as long as we find a corresponding figure in a new set, we can remove if from the new set to avoid looping and comparing definition for another figure (Not sure if it's understandable, feel free to ask in person if needed :D )